src/Engine/Animation/AnimationController.cpp

```diff
@@ -71,14 +88,37 @@ void AnimationController::Load(const std::string& name) {
         }
     }
 
+    uint32_t numBools = boolMap.size();
+    file.read(reinterpret_cast<char*>(&numBools), sizeof(uint32_t));
+    boolMap.resize(numBools);
+    for (auto i = 0; i < numBools; ++i) {
+        file.read(reinterpret_cast<char*>(boolMap[i]), sizeof(BoolItem));
+    }
+
+    uint32_t numFloats = boolMap.size();
```

Comment:
Floats doesn't have anything to do with bools. Still, this works because the value is overwritten in the next statement anyway. An idea would be to initialize to 0, which also goes for `numBools` above.